docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+
+Apache Calcite is a highly customizable engine for parsing and planning queries on relational data from various data sources. Apache Calcite provides the functionality of storage independent query optimization and ways to integrate them into other frameworks which would like to take advantage and expose SQL capability to their users. For details about Apache Calcite, please read at [Apache Calcite Website](https://calcite.apache.org/). 
+
+Particularly in SQL on Apex, Calcite processes query and then creates relational algebra to create processing pipelines. These relational algebra processing pipelines are converted to a DAG with a set of operators to perform business logic on streaming data.
+
+![image alt text](images/calcite-apex.png)
+
+Above figure explains how SQL query gets converted to Apex DAG.
+
+1. The received query from a user is parsed by Calcite’s query parser.
+2. The query parser returns the SQL Node Tree to _Query Planner & Optimizer_ which converts the SQL Node Tree to Relation Expression Tree.
+3. This Relation Expression Tree is received by Apache Apex’s SQL module to finally convert to an Apex DAG having series of operators.
+
+One peculiarity that can be seen in above example is that the data source and data destination are not RDBMS systems. This is because Calcite allows Apex to register table sources and destinations as anything which can return a row type results. So a “scan” relational expression gets converted to “KafkaInputOperator + ParseOperator”, a result of which is series of POJOs reflecting a Row Type. Similarly, the “insert” Relational Expression translated to “FormatOperator + FileOutputOperator”.
+
+For Apex-Calcite integration more details, click [here](https://github.com/apache/apex-malhar/tree/master/sql).
+# SQL APIs for Apache Apex
+
+Listed below are the Java APIs which can be used by SQL/Apex users to create a DAG in _populateDAG_ API of a streaming application.
+
+| API | Description |
+|-----|:-----------:|
+|**SQLExecEnvironment.getEnvironment()**| Creates a new SQL execution environment  |
+|**SQLExecEnvironment.registerTable(tableName, endpointInstance)**| Registers a new abstract table with existing environment. _endpointInstance_ is an object of type [Endpoint](https://github.com/apache/apex-malhar/blob/master/sql/src/main/java/org/apache/apex/malhar/sql/table/Endpoint.java) which defines a table.  |
+|**SQLExecEnvironment.registerFunction(sqlFunctionName, holderClass, staticFunctionName)**| Registers a new User Defined Scalar function |
+|**SQLExecEnvironment.executeSQL(dag, sqlStatement)** | Creates a DAG for a particular SQL statement |
+
+Usage of above APIs is described in details in following sections.
+
+## Example 1: Pure Style SQL Application
+
+With Apache Calcite Integration, you can use SQL queries across different data sources and provide UDFs as per your business logic. This example will use Kafka as input source and HDFS file as the output destination.
```

Comment:
The UDF reference seems cryptic, suggest adding "(User Defined Functions)" but if SQL users would commonly know what that means, you can leave it as is.
"will use Kafka as input source" => will use a Kafka topic as the source
"HDFS file as the output destination" => a HDFS file as the destination